DigitalExcellence/dex-backend
# Feature/256 notification system

## Description
- Added initial version of the notification system using RabbitMQ as a message broker. 
- Unit and integration tests are going to be added in a separate issue

## Type of change


## Checklist


## Steps to Test or Reproduce

For a working instance of the notification service you should create a `.env` file in the root of the project with the following environmental variables:
```
RABBITMQ_HOST_NAME=rabbitmq    #if you run it from docker-compose use rabbitmq otherwise localhost for development environment
RABBITMQ_USERNAME=
RABBITMQ_PASSWORD=
SENDGRID_API_KEY=
SENDGRID_EMAIL_FROM=
```


In order to use the notification system in the API use the following snippet:
```
NotificationSender sender = new NotificationSender();
EmailNotification notification = new EmailNotification("test@example.com", "plain text content");
sender.RegisterNotification(Newtonsoft.Json.JsonConvert.SerializeObject(notification), Subject.EMAIL);
```


## Link to issue

Closes: #256
